src/components/Root/index.tsx

```diff
@@ -33,6 +36,10 @@ const RootConsumer = (): React.ReactElement | null => {
     const initChains = async () => {
       try {
         await loadChains()
+        if (!isValidChainId(_getChainId())) {
```

Comment:
Will this check happen after the chainId is set from the URL?

Reply:
Yes, seemingly so. @DiogoSoaress is working on an extension of this https://github.com/gnosis/safe-react/issues/3218, which we just successfully tested.